Upgrade to use latest versions of zarr

### Feature Description

zarr version 3.0.0+ introduces a breaking change, causing
TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
errors in 7 test cases of tests/test_reader.py
This error traces back to the tifffile dependency, and is not an issue the maintainers of tifffile have on their calendar.

Ideally, our code should be aware of the possible upgrades and work with them rather than just pin away from them.
Why do we have a zarr dependency in a tiff library anyway?

### Use Case

As a user, I'd like to be able to use the latest modules with my code

### Solution

bioio-tiff-glob should detect what version of zarr is installed, and use the correct contract for that version

### Alternatives

Leave the code pinned to <3.0.0

With python 3.11.11, when we upgrade the zarr library, dask mismatches cause the tests to fail:
============================================================================================================= short test summary info ==============================================================================================================
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_reader_2d - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_types[list] - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_types[Series] - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_types[array] - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_mm_indexer - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_reader_3d - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
FAILED bioio_tiff_glob/tests/test_reader.py::test_glob_reader_4d - TypeError: Unsupported type for store_like: 'ZarrFileSequenceStore'
===================================================================================================== 7 failed, 1 passed, 1 xfailed in 13.01s ======================================================================================================

bioio_tiff_glob/tests/test_reader.py:200:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.11/site-packages/bioio_base/reader.py:429: in xarray_dask_data
    self._xarray_dask_data = self._read_delayed()
bioio_tiff_glob/reader.py:357: in _read_delayed
    darr = da.from_zarr(zarr_im).rechunk(-1)
.venv/lib/python3.11/site-packages/dask/array/core.py:3807: in from_zarr
    z = zarr.open_array(store=url, path=component, **kwargs)
.venv/lib/python3.11/site-packages/zarr/api/synchronous.py:1068: in open_array
    sync(
.venv/lib/python3.11/site-packages/zarr/core/sync.py:163: in sync
    raise return_result
.venv/lib/python3.11/site-packages/zarr/core/sync.py:119: in _runner
    return await coro
.venv/lib/python3.11/site-packages/zarr/api/asynchronous.py:1246: in open_array
    store_path = await make_store_path(store, path=path, mode=mode, storage_options=storage_options)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,10 +33,9 @@ dependencies = [
   "fsspec>=2022.8.0",
   "numpy>=1.21",
   "pandas>=1.1.3",
-  "tifffile>=2022.4.22",
+  "tifffile[zarr]>=2022.4.22",
   "xarray>=0.16.1",
   #zarr pinned since breaking changes in 3.0 are available with python 3.11
-  "zarr>=2.6,<3.0.0", # for tifffile
 ]
 
 [project.urls]

